icu4c/source/i18n/dtptngen.cpp

```diff
@@ -1648,7 +1648,7 @@ DateTimePatternGenerator::adjustFieldTypes(const UnicodeString& pattern,
                          (typeValue==UDATPG_MINUTE_FIELD && (options & UDATPG_MATCH_MINUTE_FIELD_LENGTH)==0) ||
                          (typeValue==UDATPG_SECOND_FIELD && (options & UDATPG_MATCH_SECOND_FIELD_LENGTH)==0) ) {
                          adjFieldLen = field.length();
-                    } else if (specifiedSkeleton) {
+                    } else if (specifiedSkeleton && reqFieldChar != LOW_C && reqFieldChar != LOW_E) {
```

Comment:
Please modify the comments between line 1618 - 1640 to reflect the behavior after your change.

Reply:
That seems like the wrong place to document this change; all the text there has to do with the hour characters, and this change only affects the day-of-week characters.  I added a new comment right above my change.

Reply:
I somehow didn't see the " a new comment right above my change." in this PR.

Reply:
You're right-- it looks like I forgot to push.  I've pushed the new fix now.  Sorry about that...